x/rollapp/keeper/hard_fork.go

```diff
@@ -195,3 +199,18 @@ func (k Keeper) pruneFinalizationsAbove(ctx sdk.Context, rollappID string, lastS
 	}
 	return nil
 }
+
+// is forking to the latest height going to violate assumptions?
+func (k Keeper) ForkLatestAllowed(ctx sdk.Context, rollapp string) bool {
+	lastHeight, ok := k.GetLatestHeight(ctx, rollapp)
+	if !ok {
+		return false
+	}
+	return k.ForkAllowed(ctx, rollapp, lastHeight)
+}
+
+// is the rollback fork going to violate assumptions?
+func (k Keeper) ForkAllowed(ctx sdk.Context, rollapp string, lastValidHeight uint64) bool {
+	ra := k.MustGetRollapp(ctx, rollapp)
+	return 0 < ra.GenesisState.TransferProofHeight && ra.GenesisState.TransferProofHeight <= lastValidHeight
```

Comment:
missing `true` on `ra.GenesisState.TransferProofHeight == 0`

Reply:
we're not gonna allow to fork before genesis bridge period (regardless of if ibc opened or not) to avoid now edge cases.

Reply:
isn't it the use case this PR tries to solve?
we want to allow fork **before** genesis bridge as well
we just want to avoid rollback from `post GB` to `pre GB`

Reply:
No @mtsitrin we are not